main.go

```diff
@@ -264,15 +264,11 @@ func processConfig(config *Config) (include *regexp.Regexp, exclude *regexp.Rege
 		config.Sort = []string{"path"}
 	}
 
-	// PlaceHolder to skipping "vendor" directory if GO15VENDOREXPERIMENT=1 is enabled.
-	// TODO(alec): This will probably need to be enabled by default at a later time.
-	if os.Getenv("GO15VENDOREXPERIMENT") == "1" || config.Vendor {
-		if err := os.Setenv("GO15VENDOREXPERIMENT", "1"); err != nil {
-			warning("setenv GO15VENDOREXPERIMENT: %s", err)
-		}
-		config.Skip = append(config.Skip, "vendor")
-		config.Vendor = true
+	if err := os.Setenv("GO15VENDOREXPERIMENT", "1"); err != nil {
+		warning("setenv GO15VENDOREXPERIMENT: %s", err)
 	}
+	config.Skip = append(config.Skip, "vendor")
```

Comment:
Isn't the standard go tooling behaviour to only skip this if paths == `./...` ?

Reply:
According to `go help packages`:

> To make common patterns more convenient, there are two special cases.
> First, /... at the end of the pattern can match an empty string,
> so that net/... matches both net and packages in its subdirectories, like net/http.
>Second, any slash-separated pattern element containing a wildcard never
>participates in a match of the "vendor" element in the path of a vendored
> package, so that ./... does not match packages in subdirectories of
> ./vendor or ./mycode/vendor, but ./vendor/... and ./mycode/vendor/... do.
> Note, however, that a directory named vendor that itself contains code
> is not a vendored package: cmd/vendor would be a command named vendor,
> and the pattern cmd/... matches it.

So it's broader than only `./...`, but indeed the behaviour we get from this won't allow us to do something like `gometalinter vendor/...` it appears. It does only occur when there is a `...` wildcard though.

Reply:
I should mention that this problem doesn't exist in v3-unstable as it uses the standard third party expansion library instead of custom code.

Reply:
Specifically github.com/kisielk/gotool

Reply:
Hmm, maybe we could backport that here then. Or is that too large a change do you think?